Rename priority to queue in the context of torrents

thalieht opened the pull request:
The one thing that troubled me particularly is the labels in the toolbar. They are kind of nonsensical but i can't think of anything else that is short enough. I also added more explanatory tooltips for them.
![no-move](https://user-images.githubusercontent.com/6451685/49744158-e0229900-fca4-11e8-91e3-ebc5162b8d7a.JPG)

I stayed clear from WebUI/API because i'm not touching the API and it would probably be confusing if half the things said queue and the other half priorities.

Closes #2514, closes #4567, closes #9910

Piccirello commented:
Can you update the WebUI too? The API doesn't have to change, but I don't want to have to create a separate PR now just to keep the GUI and WebUI consistent.

thalieht commented:
@Piccirello whew it was harder than i thought, mostly because i don't know js and HTML but i think i did most of it. There are a few things that i don't know if they're changeable:

I can't find where these are used so i can't change them
https://github.com/qbittorrent/qBittorrent/blob/e1f19b7c7531b4d3bd1b0b83e84302c42c77f210/src/webui/api/torrentscontroller.h#L68-L71
I don't know how to change "topPrioLink" and the others
https://github.com/qbittorrent/qBittorrent/blob/e1f19b7c7531b4d3bd1b0b83e84302c42c77f210/src/webui/www/private/index.html#L56-L59
I suppose this is API related because if i change it, it shows nothing in the queue column
https://github.com/qbittorrent/qBittorrent/blob/e1f19b7c7531b4d3bd1b0b83e84302c42c77f210/src/webui/www/private/scripts/dynamicTable.js#L784

Let me know if i've done something i shouldn't.

Piccirello commented:
Tested the WebUI and this works great. Thanks for making the change.

> I can't find where these are used so i can't change them

Those translate to the actual endpoints in the Web API. So `TorrentsController::increasePrioAction()` translates to `/api/v2/torrents/increasePrio`. Probably not worth changing them.

> I don't know how to change "topPrioLink" and the others

The current JS implementation makes it difficult to change the link name because it also uses it as the API name. Probably fine to leave these as is. If we do want to change the API then we can easily change all of these at once.

glassez commented on src/gui/mainwindow.ui:
```diff
@@ -248,33 +248,45 @@
+  <action name="actionBottomQueuePos">
    <property name="text">
-    <string>Minimum Priority</string>
+    <string>Bottom Queue</string>
```
IMO, "Bottom Position" and so on...

LordNyriox replied:
@thalieht, @glassez:  "Position" seems pretty generic, but… 

This one is somewhat tricky, because there is no adjective form (or standard adjective terminology) for the English word `queue`.  

The shortest formal phrasing I can think of (involving "queue") is: 

>Top of Queue 
>Raise in Queue 
>Lower in Queue 
>Bottom of Queue

thalieht replied:
This is not much better IMO because one may wonder Bottom Position of what? Unless we rely on the tooltip. Grammatically it's definitely better though. I'll wait a while in case there is some better proposition.

thalieht replied:
![capture4](https://user-images.githubusercontent.com/6451685/49791140-794ebf80-fd38-11e8-95a2-460516c68076.JPG)
I suppose i could get rid of the different tooltips with this because these are pretty self explanatory.

glassez commented on src/gui/transferlistsortmodel.cpp:
```diff
@@ -152,10 +152,10 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
-            const int prioR = model->data(model->index(right.row(), TransferListModel::TR_PRIORITY)).toInt();
-            const bool isSeedingL = (prioL < 0);
-            const bool isSeedingR = (prioR < 0);
+            const int queueL = model->data(model->index(left.row(), TransferListModel::TR_QUEUE_POSITION)).toInt();
```
posL or queuePosL

thalieht replied:
Done.

glassez commented on src/gui/transferlistwidget.cpp:
```diff
@@ -553,12 +553,12 @@ void TransferListWidget::copySelectedHashes() const
-    setColumnHidden(TransferListModel::TR_PRIORITY, hide);
-    if (!hide && !columnWidth(TransferListModel::TR_PRIORITY))
-        resizeColumnToContents(TransferListModel::TR_PRIORITY);
+    qDebug("hideQueuePosColumn(%d)", hide);
```
Please remove it.

LordNyriox replied:
@glassez:  The whole "hide queue column" handler, or just the debug line?

glassez replied:
debug line

thalieht replied:
OK

thalieht commented:
Squashed GUI and WebUI commits, removed different tooltip text from labels (will undo if you liked it) and changed labels as per @LordNyriox's suggestion.

LordNyriox commented:
@thalieht:  

>removed different tooltip text from labels (will undo if you liked it) 

Please restore the tooltips.  They will be easier to translate in a legible way—and are slightly more clear than the labels to boot. 

I too am dissatisfied with how the button labels are phrased, but tooltips could certainly help keep things clear.  In fact, I think we should add tooltips whenever the behavior of a button or option might seem unclear to a new user. 

**Rant:**  On the topic of tooltips, I would frankly like an explanatory tooltip for every `Advanced Option` as well (including some indicator of what the range of **acceptable** values are, not just the default ones).  

I do not want to go hunting through Libtorrent-rasterbar's poorly-maintained documentation to figure out what a single option does—only to hunt through 10 more pages of miscellaneous documentation and actual code, just to understand the answer.

LordNyriox commented:
@thalieht: 

>I would be willing to make a PR adding tooltips but *not* willing to create them. 

Please do create that PR.  You can mark it as `[WIP]`, and attach a request for devs to suggest the actual tooltips.  (Similar to the discussion on the button labels in this PR.) 

From there, you could add a commit for each tooltip as it is suggested.  Please?

Chocobo1 commented:
The changes looks alright, if nothing else you can squash the last commit.

glassez commented on src/gui/transferlistwidget.cpp:
```diff
@@ -553,12 +553,11 @@ void TransferListWidget::copySelectedHashes() const
-    if (!hide && !columnWidth(TransferListModel::TR_PRIORITY))
-        resizeColumnToContents(TransferListModel::TR_PRIORITY);
+    setColumnHidden(TransferListModel::TR_QUEUE_POSITION, hide);
+    if (!hide && !columnWidth(TransferListModel::TR_QUEUE_POSITION))
```
Please fix it along the way. `!columnWidth()` => `columnWidth() == 0`

glassez commented on src/webui/www/private/transferlist.html:
```diff
@@ -42,17 +42,17 @@
             },
-            prioTop: function(element, ref) {
-                setPriorityFN('topPrio');
+            QueueTop: function(element, ref) {
```
Why it starts with Capital letter?

thalieht replied:
I followed the style of the other ones.

glassez replied:
However, this style does not follow the style of the project... Okay, whatever. I don't care.

thalieht replied:
Doesn't sound like you don't care but there's not much i can do about it because i can't even tell which ones are the vars.

Chocobo1 replied:
You should use lowercase for new code.

thalieht replied:
Added new commit to rename some of them.

glassez commented:
@Piccirello, can you review WebUI part of this PR?

thalieht commented on src/webui/www/private/scripts/dynamicTable.js:
```diff
@@ -948,8 +948,8 @@ var TorrentsTable = new Class({
 
         // priority
         this.columns['priority'].updateTd = function(td, row) {
```
Wait i'll revert the `queue_position` because i can't change it in https://github.com/qbittorrent/qBittorrent/blob/e1f19b7c7531b4d3bd1b0b83e84302c42c77f210/src/webui/www/private/scripts/dynamicTable.js#L784
Also found something else that i'll mention when i push in a few minutes.

thalieht replied:
Ok rebased, reverted `queue_position` and found this little bugger that i previously missed, everything was broken because of it https://github.com/qbittorrent/qBittorrent/pull/9995/files#diff-700c2fec148a222fce14d3faa49b7a62R68

glassez replied:
@Piccirello, is it fixed now?

Chocobo1 replied:
@thalieht 
There are still a few `'priority'` instances in this file, is it possible to replace them without breaking anything?

thalieht replied:
@Chocobo1 no.

thalieht commented:
Rebased.

thalieht commented:
Rebased. IIRC the only problem was with some vars in the API that i won't touch. Tested GUI and WebUI again and everything seems to be fine.

LordNyriox commented on src/gui/mainwindow.ui:
```diff
@@ -248,33 +248,45 @@
+  <action name="actionDecreaseQueuePos">
    <property name="text">
-    <string>Decrease Priority</string>
+    <string>Lower in Queue</string>
```
@thalieht:  Hmm.  In retrospect, I think `Drop in Queue` would read a little better, in this case.

thalieht replied:
Doesn't sound better to me but that's because for some reason i expect "from" after "Drop" when reading it fast. Sounds stupid, i know, and if you're sure i'll change it.

LordNyriox replied:
@thalieht:  I just thought "lower" seems a mouthful to say.  "Drop" means the same thing, but uses only one syllable instead of two—and four letters instead of five. 

If you truly think the shortened form will be be confusing (despite the tooltip), feel free to keep the label as `Lower`.  :}

LordNyriox replied:
@thalieht:  I just thought "lower" seems a mouthful to say.  "Drop" means the same thing, but uses only one syllable instead of two—and four letters instead of five. 

If you truly think the shortened form will be be confusing (despite the tooltip), feel free to keep the label as `Lower`.  :}

thalieht replied:
I don't have a strong opinion about this and unfortunately i don't know what other people will think of it.

thalieht replied:
@Chocobo1 @glassez @Piccirello any opinions on this?

Chocobo1 replied:
I'm fine with both. LordNyriox version does sound terser than current one.

glassez replied:
Sorry, I have no strong opinion since I'm not native English speaker. I don't understand such nuances.

Piccirello replied:
Would everyone hate it if I said “Decrease in queue” sounded better? :)

I don’t think drop is the right word- it sounds more like a pop operation.

thalieht replied:
Personally i don't like "decrease". Maybe just copy utorrent and be done with it? 
"Move Up/Down Queue"

LordNyriox replied:
@thalieht:  `Move Up Queue` / `Move Down Queue` does sound nice.  You can go ahead and use that.  :}

thalieht replied:
Changed.

Chocobo1 commented:
@thalieht 
Thank you.

thalieht commented:
Thanks everyone
